WIP: added advanced config in route

theliuk opened the pull request:
<!-- Hi, and thank you for your time dedicated to this pull request! Please provide a description
of the work you have done and be sure to link to the relative open issue if is present.

Be aware that all the work you have done should include also the relative tests to assure the
correct behavior and avoid possible regressions in the future.
-->

This PR is to support all features that fastify has.
See https://www.fastify.io/docs/latest/Routes/#full-declaration

#### Checklist

- [x] your branch will not cause merge conflict with `master`
- [x] run `npm test`
- [x] tests are included
- [x] the documentation is updated or intregrated accordingly with your changes
- [x] the code will follow the lint rules and style of the project
- [x] you are not committing extraneous files or sensible data

allevo commented:
- Test: spread operator precedence.
- TypeScript: see https://github.com/fastify/fastify/blob/9b53f9a554514f0555da4bf2e2ad5126df4fcbeb/fastify.d.ts#L252

theliuk commented:
1. I added tests to check that some part of advanced configuration is properly overwritten when passed to fastify.route.

2. I updated the signature of the **addRawCustomPlugin** method.
